app/src/main/java/org/sopt/at/views/history/components/dialog/HistoryAddDialog.kt

```diff
@@ -0,0 +1,189 @@
+package org.sopt.at.views.history.components.dialog
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.IntrinsicSize
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.Card
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.OutlinedTextFieldDefaults
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.compose.ui.window.Dialog
+import androidx.hilt.navigation.compose.hiltViewModel
+import org.sopt.at.R
+import org.sopt.at.core.common.CommonConstants
+import org.sopt.at.models.history.HistoryEntity
+import org.sopt.at.ui.theme.ATSOPTANDROIDTheme
+import org.sopt.at.viewmodels.HistoryViewModel
+import org.sopt.at.views.home.common.HomeItemTypeDescription
+import org.sopt.designsystem.theme.MyAtSoptTheme
+
+@Composable
+fun HistoryAddDialog(
+    modifier: Modifier = Modifier,
+    historyViewModel: HistoryViewModel = hiltViewModel(),
+    onConfirm: (HistoryEntity) -> Unit = {},
+    onDismiss: () -> Unit = {}
+) {
+    val historyTitle by historyViewModel.historyTitle.collectAsState()
+
+    var isError by remember {
```

Comment:
이거 lauchedeffect로 바꿀 필요없이
```suggestion
    val isError = historyTitle.isNullOrEmpty()
```

이렇게만 해도 될거같아요.